packages/keybr-textinput-events/lib/TextEvents.tsx

```diff
@@ -226,28 +226,17 @@ export class TextEvents extends PureComponent<Props> {
   }
 }
 
-function toKeyEvent(
-  {
-    code,
-    key,
-    shiftKey,
-    altKey,
-    ctrlKey,
-    metaKey,
-    location,
-    repeat,
-  }: KeyboardEvent,
-  timeStamp: number,
-): KeyEvent {
-  return {
-    timeStamp,
-    code,
-    key,
-    shiftKey,
-    altKey,
-    ctrlKey,
-    metaKey,
-    location,
-    repeat,
-  };
+/**
+ * Process keyboard event by adding timestamp.
+ * @param ev
+ * @param timeStamp
+ * @returns Original keyboard event with a timestamp
+ */
+function toKeyEvent(ev: KeyboardEvent, timeStamp: number): KeyEvent {
```

Comment:
I would leave the original `KeyEvent` interface. Then I would extend it with a property for the modifiers. Smth like this:

```
interface KeyEvent {
  // A copy of the KeyboardEvent properties.
  // ...
  /** Event occurrence time. */
  readonly timeStamp: number;
  /** The state of keyboard modifier keys. */
  readonly modifiers: readonly string[];
}
```

The `modifiers` property would keep an array of strings such as `"CapsLock"`, `"NumLock"`.

Such a design will make the remaining code below simpler.